pymoto/modules/assembly.py

```diff
@@ -245,28 +281,72 @@ class AssembleMass(AssembleGeneral):
-            weights = np.array([8.0, 4.0, 2.0, 1.0])
-            for n1 in range(domain.elemnodes):
-                for n2 in range(domain.elemnodes):
-                    dist = round(np.sum(abs(np.array(domain.node_numbering[n1]) - np.array(domain.node_numbering[n2])))/2)
-                    ME[n1*domain.dim+np.arange(domain.dim), n2*domain.dim+np.arange(domain.dim)] = weights[dist]
-
-            ME *= mel / 216
-        else:
-            raise RuntimeError("Only for 2D and 3D")
-        super()._prepare(domain, ME, *args, bcdiagval=bcdiagval, **kwargs)
+
+class AssembleScalarMass(AssembleGeneral):
+    r""" Consistent mass equivalent matrix assembly by scaling elements (e.g. Thermal Capacity)
+        :math:`\mathbf{C} = \sum_e x_e \mathbf{C}_e`
+
+        Input Signal:
+            - ``x``: Scaling vector of size ``(Nel)``
+
+        Output Signal:
+            - ``C``: Scalar mass equivalent matrix of size ``(n, n)``
+
+        Args:
+            domain: The domain to assemble for -- this determines the element size and dimensionality
+            *args: Other arguments are passed to AssembleGeneral
+
+        Keyword Args:
+            MP: Material property for element matrix (e.g. Thermal capacity)
+            bcdiagval: The value to put on the diagonal in case of boundary conditions (bc)
+            **kwargs : Other keyword-arguments are passed to AssembleGeneral
+        """
+
+    def _prepare(self, domain: DomainDefinition, *args, MP: float = 1.0, bcdiagval=0.0, **kwargs):
+        # Element mass equivalent matrix
+        self.CE = ConsistentMassEq(domain, ndof=1, MP=MP)
+        super()._prepare(domain, self.CE, *args, bcdiagval=bcdiagval, **kwargs)
+
+
+class AssembleScalarField(AssembleGeneral):
+    r"""
+    Scalar field matrix assembly (e.g. Thermal conductivity, Electric permittivity)
+    :math:`\mathbf{Ks} = \sum_e x_e \mathbf{Ks}_e`
+
+    Input Signal:
+        - ``x``: Scaling vector of size ``(Nel)``
+
+    Output Signal:
+        - ``Kt``: Scalar field matrix of size ``(n, n)``
+
+    Args:
+        domain: The domain to assemble for -- this determines the element size and dimensionality
+        args (optional): Other arguments are passed to AssembleGeneral
+
+    Keyword Args:
+        kt: Material property (e.g. Thermal conductivity, Electric permittivity)
+        bcdiagval: The value to put on the diagonal in case of boundary conditions (bc)
+        kwargs: Other keyword-arguments are passed to AssembleGeneral
+    """
+
+    def _prepare(self, domain: DomainDefinition, *args, kt: float = 1.0, **kwargs):
+        # Prepare material properties and element matrices
+        self.kt = kt
+        kappa = np.identity(domain.dim)*self.kt
+        self.SFE = np.zeros((domain.elemnodes, domain.elemnodes))
+
+        # Numerical Integration
+        siz = domain.element_size
+        w = np.prod(siz[:domain.dim]/2)
+        if domain.dim != 3:
+            kappa *= siz[domain.dim:]
```

Comment:
Add some description; what is kappa? probably called kappa as it is written in some book; but reader has not read your book.

Reply:
kappa is a way to specify different material properties for different directions, allows for anisotropic conductivity (or similar). Thinking about it I might just remove kappa as we're not doing anisotropy anywhere else.

Reply:
thanks for the explanation! however, it is not relevant whether I understand it, it is relevant whether any user/reader of the code can understand this; i.e. better add the explanation as a comment or, as you indicate, remove the kappa if not needed.